processor/src/chiplets/memory/segment.rs

```diff
@@ -70,7 +71,7 @@ impl MemorySegmentTrace {
     ///
     /// If the specified address hasn't been previously written to, four ZERO elements are
     /// returned. This effectively implies that memory is initialized to ZERO.
-    pub fn read(&mut self, addr: u32, clk: Felt) -> Word {
+    pub fn read(&mut self, addr: RowIndex, clk: Felt) -> Word {
```

Comment:
@bobbinth just checking whether `clk` here and similar place should be RowIndex or u32 like addr.
```
pub fn read(&mut self, ctx: ContextId, addr: u32, clk: RowIndex)
```

Reply:
Also whether this makes sense as RowIndex
```
pub struct MemorySegmentTrace(BTreeMap<RowIndex, Vec<MemorySegmentAccess>>);
```

Reply:
> Also whether this makes sense as RowIndex
> 
> ```
> pub struct MemorySegmentTrace(BTreeMap<RowIndex, Vec<MemorySegmentAccess>>);
> ```

Yeah - I think this should be `pub struct MemorySegmentTrace(BTreeMap<u32, Vec<MemorySegmentAccess>>);` As the key in the map is memory address rather than row index.